fs/vfs/fs_ioctl.c

```diff
@@ -67,53 +67,52 @@ int file_vioctl(FAR struct file *filep, int req, va_list ap)
       ret = inode->u.i_ops->ioctl(filep, req, arg);
     }
 
-  /* Check for File system IOCTL commands that can be implemented via
-   * fcntl()
-   */
-
-  if (ret != -ENOTTY)
-    {
-      return ret;
-    }
-
   switch (req)
     {
       case FIONBIO:
-        {
-          FAR int *nonblock = (FAR int *)(uintptr_t)arg;
-          if (nonblock && *nonblock)
-            {
-              filep->f_oflags |= O_NONBLOCK;
-            }
-          else
-            {
-              filep->f_oflags &= ~O_NONBLOCK;
-            }
-
-          ret = OK;
-        }
+        if (ret == OK || ret == -ENOTTY)
```

Comment:
Maybe just
```
if (ret == -ENOTTY)
  {
    ret = OK;
  }
```
before the `switch` and then check `if (ret == OK)` in case statements?

Reply:
Can't since this will bypass FIOC_FILEPATH and BIOC_BLKSSZGET